### DISCUSSION TITLE
Read API?

### DISCUSSION THREAD

**Discussion by rexxars** [tphakala/birdnet-go#253] [Ideas]
Hi, thanks for all your hard work!

Is there any chance you'd consider adding a (read only) REST API that exposes data as JSON?
That would make it really easy to build all sorts of custom visualizations and integrations.

**Comment by tphakala** [comment]
I have no plans for such API at this point, but I am open for contributions.

**Comment by jkrauska** [comment]
@rexxars what data would you want the API to expose?  most recent detections?

**Reply by rexxars** [reply]
Yes - ideally most recent detections and possibly a stream (maybe using [Server-Sent Events](https://html.spec.whatwg.org/multipage/server-sent-events.html)) for new detections.

That would allow for all kinds of nice dashboards/external integrations :)

**Comment by tphakala** [comment]
I started implenting JSON API as I am pivoting UI to Svelte based one soon, this will provide plenty of read methods. https://github.com/tphakala/birdnet-go/pull/500

**Reply by tphakala** [reply]
@rexxars JSON API is now available in https://github.com/tphakala/birdnet-go/releases/tag/nightly-20250228, I documentation is still pending, I need to look at swagger for automated generation.

**Reply by Webreaper** [reply]
Fantastic. I'm thinking of writing a script that queries the recent detections, and posts it to a page our village website so anyone viewing the site can see today's birds. This will be most helpful!
